homeassistant/components/icloud/config_flow.py

```diff
@@ -180,7 +180,6 @@ async def async_step_reauth(self, user_input=None):
         # Store existing entry data so it can be used later and set unique ID
         # so existing config entry can be updated
         if not self._existing_entry:
-            await self.async_set_unique_id(user_input.pop("unique_id"))
```

Comment:
Why doesn't this abort if there is more than one flow in progress for the same unique id?

Reply:
It raises an AbortFlow instead of returning one. Does that achieve the same thing? If so then this PR is unnecessary

Reply:
Yes, the abort exception will be handled by the data entry flow manager.

Reply:
I think we should take advantage of the unique id api. It makes these things easier.

Reply:
OK great, then this PR is not needed. Thanks!